src/components/ShoppingList/ShoppingList.js

```diff
@@ -168,18 +168,34 @@ function ShoppingList({ listId, handleModalOpen }) {
       item.numberOfPurchases,
     );
 
-    db.collection(`lists/${listId}/items`)
+    return db
+      .collection(`lists/${listId}/items`)
       .doc(item.id)
       .update({
-        lastPurchaseDate: new firebase.firestore.Timestamp(
-          newPurchaseDate.toSeconds(),
-          0,
-        ),
+        lastPurchaseDate: firebase.firestore.FieldValue.serverTimestamp(),
         numberOfPurchases: firebase.firestore.FieldValue.increment(1),
         purchaseInterval: newPurchaseInterval,
-      })
-      .catch((err) => {
-        console.log(err);
+        // back up some info in case user mistakenly checks item and wants to undo
+        undoRestore: {
```

Comment:
nit: I'm not sure this variable name conveys well what it's purpose is.  Perhaps something like "backupValues" or "originalValues" might read a bit better?

Reply:
Actually, I agree. I like backupValues and will switch that out before merging.